### PR TITLE
More use of HtmlString

### PR DESCRIPTION
**Pull request by labkey-adam**
#39589, #39590

### CODE REVIEW

File: api/src/org/labkey/api/jsp/taglib/InputTag.java [hunk 1]
```diff
@@ -215,9 +216,26 @@ public void setType(String type)
...
+        this.value = hasHtmlString.getHtmlString();
+    }
+
+    @Deprecated // TODO: Just call the String version without h(), like all the other tag attributes
+    public void setValue(HtmlString htmlString)
+    {
+        this.value = htmlString;
+    }
+
+    public void setValue(String s)
```
Comment (labkey-jeckels) [inline api/src/org/labkey/api/jsp/taglib/InputTag.java:233]: Suggest adding JavaDoc that this will be HTML-encoded

**Review by labkey-jeckels — APPROVED** [review]
One suggested comment to add to the code.

Do these <%= x %> usages in JSPs cause warnings about not encoding? Do we need to update our rules?

File: api/src/org/labkey/api/jsp/taglib/InputTag.java (hunk 1 above)
Comment (labkey-adam) [reply api/src/org/labkey/api/jsp/taglib/InputTag.java:233]: Done

**Comment by labkey-adam** [comment]
Yes, our convention for tags is to not encode the "attributes"; this tag was an outlier, which is what caused problems. We still want squiggles on most raw strings in JSPs, but I can't figure out how to get the template to ignore attributes within tags.

**Comment by labkey-adam** [comment]
Good news is that the tag now correctly handles both encoded and unencoded content, so we're safe either way.

**Pull request merged**